include/zio/zio_defs.h

```diff
@@ -0,0 +1,510 @@
+ * | | Channel 0---------------------------------------------------------+ | |
+ * | | | Channel Attributes---------------------------------------------+| | |
+ * | | | | Channel Attribute 0----------------------------------------+ || | |
+ * | | | | |                   CHAN_ATTR_RAW_DATA                     | || | |
+ * | | | | Channel Attribute 1----------------------------------------+ || | |
+ * | | | | |                   CHAN_ATTR_SI_DATA                      | || | |
+ * | | | | Channel Attribute 2----------------------------------------+ || | |
+ * | | | | |                   CHAN_ATTR_SAMP_FREQ                    | || | |
+ * | | | | +----------------------------------------------------------+ || | |
+ * | | | +--------------------------------------------------------------+| | |
+ * | | +-----------------------------------------------------------------+ | |
+ * | | |Channel 1                                                        | | |
+ * | | +-----------------------------------------------------------------+ | |
+ * | | |Channel 2                                                        | | |
+ * | | +-----------------------------------------------------------------+ | |
+ * | +---------------------------------------------------------------------+ |
+ * +-------------------------------------------------------------------------+
+ */
+
+/**
+ * @brief ZIO device channel types.
+ *
+ * Each channel type is associated with a specific SI unit type, described
+ * in the inline comments for each enum entry.
+ */
+typedef enum zio_dev_chan_type {
+	/**
+	 * Raw data channel.
+	 *
+	 * No SI type is prescribed for this channel.
+	 */
+	DEV_CHAN_RAW,
+
+	/**
+	 * Voltage.
+	 *
+	 * Expressed as volts (V).
+	 */
+	DEV_CHAN_VOLTAGE,
+
+	/**
+	 * Current.
+	 *
+	 * Expressed as milli-amps (mA).
+	 */
+	DEV_CHAN_CURRENT,
+
+	/**
+	 * Resistance.
+	 *
+	 * Expressed as ohms.
+	 */
+	DEV_CHAN_RESISTANCE,
+
+	/**
+	 * Capacitance.
+	 *
+	 * Expressed as picofarad (pF).
+	 */
+	DEV_CHAN_CAPACITANCE,
+
+	/**
+	 * Inductance.
+	 *
+	 * Expressed as microhenries (uH).
+	 */
+	DEV_CHAN_INDUCTANCE,
+
+	/**
+	 * Impedance.
+	 *
+	 * Expressed as ohms (Z).
+	 */
+	DEV_CHAN_IMPEDANCE,
+
+	/**
+	 * Frequency.
+	 *
+	 * Expressed as Hertz (Hz).
+	 */
+	DEV_CHAN_FREQUENCY,
+
+	/**
+	 * Amplitude.
+	 *
+	 * Note: While amplitude is a unitless value with a floating-point
+	 * range of -1.0 .. 1.0, you can associate this with a range of integer
+	 * values by means of the 'CHAN_ATTR_RAW_DATA' attribute. Normalised
+	 * floating-point values should be assigned to 'CHAN_ATTR_SI_DATA',
+	 * but the source integer values, such as 's16_t', can be provided
+	 * as raw data to avoid expensive floating-point operations if
+	 * desired.
+	 *
+	 * Expressed as a unitless value in the range of -1.0 to 1.0.
+	 */
+	DEV_CHAN_AMPLITUDE,
+
+	/**
+	 * Generic temperature.
+	 *
+	 * Expressed as degrees celsius (C).
+	 */
+	DEV_CHAN_TEMP,
```

Comment:
Any reason to have 3 types of temperature?
Aren't DIE and AMBIENT enough?
I'm not against it. Just want to understand the reason.
Maybe we could have only two types: AMBIENT and GENERAL. The DIE case would fall into the GENERAL case.

Reply:
Many sensors out there include temperature measurement, but it's often only of the die so of less use to the outside world, and I guess I just wanted to clearly distinguish die temp (which could me many degrees different than ambient temp or the same measurement on another IC), from ambient which is device nuetral and closer to what people want, without ignore the need to sometimes track die temp for either generating appropriate offets for the sensor data, or for safety reasons.

The third (first?) generic temp entry -- `DEV_CHAN_TEMP` -- is basically for anything that isn't the two most common cases of DIE or AMBIENT, and you could then define exactly what that is via attributes (water temperature, heating-element temperature, whatever).

I'm open to change this, just trying to explain the reasoning in my own head!

Reply:
> Many sensors out there include temperature measurement, but it's often only of the die so of less use to the outside world, and I guess I just wanted to clearly distinguish die temp (which could me many degrees different than ambient temp or the same measurement on another IC), from ambient which is device nuetral and closer to what people want, without ignore the need to sometimes track die temp for either generating appropriate offets for the sensor data, or for safety reasons.

Yes, I agree.
Actually if you see the history of commits of the current sensor.h you'll see that I'm the one that separated the two temperature types. So, yes, I agree with this part.



> The third (first?) generic temp entry -- `DEV_CHAN_TEMP` -- is basically for anything that isn't the two most common cases of DIE or AMBIENT, and you could then define exactly what that is via attributes (water temperature, heating-element temperature, whatever).
> 
> I'm open to change this, just trying to explain the reasoning in my own head!

I was in fact more questioning about this third type.
We can keep it, I'm not against it. Maybe it is more clear like that, but we cannot have as many
temperatures definition as are out there. So, I was thinking to have only two types: AMBIENT and  TEMP which means other non ambient. But, again, maybe this way it is clearer.